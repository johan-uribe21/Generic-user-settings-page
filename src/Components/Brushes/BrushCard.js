import React from 'react';

// Display Component, brush card
const BrushCard = ( {color, speed, autoOff} ) => {

  return (
    <article className="center mw5 mw6-ns br3 hidden ba b--black-10 mv4">
      <h1 className="f4 bg-near-white br3 br--top black-60 mv0 pv2 ph3">User preferences</h1>
      <div className="pa3 bt b--black-10">
        <ul className=" f6 f5-ns lh-copy measure">
          <li>Color: {color}   <button className={`beam-brush-${color} box`}></button> </li>
          <li>Speed: {speed} </li>
          <li>Auto Off: {autoOff}</li>
        </ul>
      </div>
    </article>
  )
}

export default BrushCard;