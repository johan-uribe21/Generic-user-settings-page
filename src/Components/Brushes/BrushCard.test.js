import {shallow, configure} from 'enzyme';
import BrushCard from './BrushCard';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() }); 

const mockBrushList = [
    {
      id: 8, 
      member_id: 43,
      brush_color: "pink",
      motor_speed: 1.0,
      auto_off: 'No' 
      // this actually returns false, but is converted to "No" by ternary operator in the Brushes component
    }
  ]

const wrapper = shallow(<BrushCard 
    color={mockBrushList[0].brush_color} 
    speed={mockBrushList[0].motor_speed}
    autoOff={mockBrushList[0].auto_off}/>)

describe('Brush preference card component renders with the props correctly passsed',() => {
  it('expect to render individual brush card', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('expect brush card to contain button-icon with correct color', () => {
    expect(wrapper.containsMatchingElement(<button className={`beam-brush-pink box`}></button>))
      .toBeTruthy();
  });

  it('expect brush card to display Motor Speed', () => {
    expect(wrapper.containsMatchingElement(<li>Motor Speed: 1 </li>))
    .toBeTruthy();
  });

  it('expect brush card to display Auto Off with Yes or No', () => {
    expect(wrapper.containsMatchingElement(<li>Auto Off: No</li>))
      .toBeTruthy();
  });
})
