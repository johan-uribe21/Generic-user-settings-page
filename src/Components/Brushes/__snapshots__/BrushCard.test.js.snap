// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brush preference card component renders with the props correctly passsed expect to render individual brush card 1`] = `ShallowWrapper {}`;
